### PR TITLE
fix(service): set template version and metadata correctly

### PR DESCRIPTION
**Pull request by Panaetius**
Sets template version, immutable_files and automated_update correctly when creating a project through the service.

Fixes bug where `progress_callback` was instead interpreted as `check_only` on template/docker_update methods.

Adds support for template update handling template files deleted by the user.

Fixes `renku check-immutable-template-files` command.

Adds additional tests

### CODE REVIEW

**Review by jsam — APPROVED** [review]
LGTM. Thank you :rocket:

File: renku/cli/__init__.py [hunk 1]
```diff
@@ -196,6 +196,7 @@ def help(ctx):
 cli.add_command(init_command)
 cli.add_command(log)
 cli.add_command(migrate)
+cli.add_command(check_immutable_template_files)
```
Comment (mohammad-sdsc) [inline renku/cli/__init__.py:199]: We should add `hidden=True` in `migrate.py` for `check_immutable_template_files` to hide it from normal help output.

**Review by mohammad-sdsc — COMMENTED** [review]
LGTM, thanks! Just a minor comment.

File: renku/cli/__init__.py (hunk 1 above)
Comment (Panaetius) [reply renku/cli/__init__.py:199]: 👍 
Done

**Review by mohammad-sdsc — APPROVED** [review]
Thanks!

**Pull request merged**